piccolo_api/change_password/endpoints.py

```diff
@@ -0,0 +1,181 @@
+                styles=self._styles,
+                **template_context,
+            )
+        )
+
+    async def get(
+        self, request: Request
+    ) -> t.Union[HTMLResponse, PlainTextResponse]:
+        session_user = (
+            await SessionsBase.select(SessionsBase.user_id)
+            .where(SessionsBase.token == request.cookies.get("id"))
+            .first()
+        )
+        if session_user:
+            return self.render_template(request)
+        else:
+            return PlainTextResponse("Not authenticated", status_code=401)
+
+    async def post(self, request: Request) -> Response:
+        # Some middleware (for example CSRF) has already awaited the request
+        # body, and adds it to the request.
+        body = request.scope.get("form")
+
+        if not body:
+            try:
+                body = await request.json()
+            except JSONDecodeError:
+                body = await request.form()
+
+        old_password = body.get("old_password", None)
+        new_password = body.get("new_password", None)
+        confirm_password = body.get("confirm_password", None)
+
+        if (not old_password) or (not new_password) or (not confirm_password):
+            raise HTTPException(
+                status_code=401,
+                detail="Form is invalid. Missing one or more fields.",
+            )
+
+        if len(new_password) < 6:
+            if body.get("format") == "html":
+                return self.render_template(
+                    request,
+                    template_context={
+                        "error": "Password must be at least 6 characters long."
+                    },
+                )
+            else:
+                raise HTTPException(
+                    status_code=401,
+                    detail="Password must be at least 6 characters long.",
+                )
+
+        if confirm_password != new_password:
+            if body.get("format") == "html":
+                return self.render_template(
+                    request,
+                    template_context={"error": "Passwords do not match."},
+                )
+            else:
+                raise HTTPException(
+                    status_code=401, detail="Passwords do not match."
+                )
+
+        session_user = (
+            await SessionsBase.select(SessionsBase.user_id)
+            .where(SessionsBase.token == request.cookies.get("id"))
+            .first()
+        )
+
+        await self._auth_table.update_password(
+            user=session_user["user_id"], password=new_password
+        )
+
+        # after password changes we invalidate session and redirect user
+        # to login endpoint to login again with new password
+        response = RedirectResponse(
+            url="/login/", status_code=HTTP_303_SEE_OTHER
+        )
+        response.delete_cookie("id")
+        return response
+
+
+def change_password(
+    auth_table: t.Optional[t.Type[BaseUser]] = None,

```

Comment:
We can probably get rid of this - I don't think we use it.

Reply:
You're right. I will remove that. Thanks